tests/models/test_default_evaluator.py

```diff
@@ -15,11 +22,13 @@
     _get_classifier_per_class_metrics,
     _gen_classifier_curve,
     _evaluate_custom_metric,
+    _load_custom_metric_artifact,
 )
 import mlflow
 from sklearn.linear_model import LogisticRegression
 
 # pylint: disable=unused-import
+from mlflow.utils.file_utils import TempDir
```

Comment:
Can we use pytest's `tmpdir` fixture instead?
https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmpdir-fixture

Reply:
I think the usage of `tmpdir` [is no longer recommended](https://py.readthedocs.io/en/latest/path.html#:~:text=Note%3A%20The%20%E2%80%98py%E2%80%99%20library%20is%20in%20%E2%80%9Cmaintenance%20mode%E2%80%9D%20and%20so%20is%20not%20recommended%20for%20new%20projects.%20Please%20check%20out%20pathlib%20or%20pathlib2%20for%20path%20operations.) since it is based on a `py.path` object. Also, `TempDir` is a parameter that the new function `_load_custom_metric_artifact` expects from calls from an `DefaultEvaluator` object. I think I could use a [`tmp_path`](https://docs.pytest.org/en/6.2.x/tmpdir.html#the-tmp-path-fixture) fixture to provide a separate temporary path for the files the tests generate. And that will co-exist with the import of `TempDir` from MLflow.

Reply:
@MarkYHZhang Thanks for the explanation! I didn't know `tmpdir` is no longer recommended. Using `tmp_path` sounds good to me.